cloudsim-plus/src/main/java/org/cloudbus/cloudsim/vms/Vm.java

```diff
@@ -76,6 +79,35 @@
      */
     CloudletScheduler getCloudletScheduler();
 
+    /**
+     * Gets the free pes number.
+     *
+     * @return the free pes number
+     */
+    long getFreePesNumber();
+
+    /**
+     * Sets the free pes number.
+     *
+     * @return the new free pes number
+     */
+    Vm setFreePesNumber(long freePes);
```

Comment:
Actually after rethinking, I think we should keep these setters in the Vm interface. Because it is not necessary that the actual object is always going to be of VmSimple class, as in future there might be other Vm implementations.

Reply:
On the other hand, the new implementations can always extend the simple one. So what do you think? Should I still move the set methods from Vm to VmSimple?

Reply:
But the intention is that new implementations are based on existing ones, as it occurs with VmAllocationPolicy, CloudletScheduler and other classes that have an abstract implementation to be based on. The best thing to do would be creating a VmAbstract class, but I think it's too much for the current PR.

Reply:
I think, for now, moving the methods to VmSimple is the best thing to do, while the VmAbstract is not introduced in the future.

Reply:
Alright.

Reply:
But now I am getting ClassCastException [here](https://github.com/humaira-salam/cloudsim-plus/blob/443c648a5466b8633bfbd558a88d7ca07dddd9c0/cloudsim-plus/src/main/java/org/cloudbus/cloudsim/brokers/DatacenterBrokerAbstract.java#L925) because the mapper sometimes returns Vm.NULL

For now I will introduce a check if(mappedVm != Vm.NULL), or?

Reply:
Oh, I forgot about this. The Null Pattern implementation has this casting issue. 
Just move inside the if.

Reply:
Done.